### PR TITLE
feat: Add Blake2s benchmark program

### PR DESCRIPTION
**Pull request by storojs72**
Adds simple Sphinx program that performs single blake2s hashing, useful for testing and benchmarking the correspondent precompile (https://github.com/argumentcomputer/sphinx/pull/176). 

The integration code is located [here](https://github.com/argumentcomputer/RustCrypto-hashes/blob/zkvm/blake2/src/macros.rs#L146).

This PR is better to merge after https://github.com/argumentcomputer/sphinx/pull/176 with correspondent update to Cargo.toml. 

Nit: In order to compile the programs (run `make`), it is necessary to downgrade `hybrid-array` dependency in every program in `programs` and also in the kadena workspace (hardcoding 0.2.0-rc.9 version in Cargo.toml didn't help me to workaround this):
```
cargo update hybrid-array@0.2.0-rc.10 --precise 0.2.0-rc.9
```

### CODE REVIEW

File: kadena/core/src/crypto/hash/sha512.rs [hunk 1]
```diff
@@ -3,11 +3,11 @@
 
 use crate::crypto::error::CryptoError;
 use crate::crypto::hash::HashValue;
-use sha2::{Digest, Sha512Trunc256};
+use sha2::{Digest, Sha512_256};
 
 // Hash functions for Merkle tree nodes
 // cf. https://github.com/kadena-io/chainweb-node/wiki/Chainweb-Merkle-Tree#merke-log-trees
-pub type ChainwebHash = Sha512Trunc256;
+pub type ChainwebHash = Sha512_256;
```
Comment (wwared) [inline kadena/core/src/crypto/hash/sha512.rs:10]: Is this change necessary?

Comment (storojs72) [reply kadena/core/src/crypto/hash/sha512.rs:10]: From my understanding, `Sha512Trunc256` is relevant for the [0.9](https://github.com/RustCrypto/hashes/blob/backports/digest-v0.9/sha2/src/lib.rs) version of `sha2` crate which is quite outdated, while our zkvm branch contains more recent version (10.0.8) with specifically [exposed types](https://github.com/argumentcomputer/RustCrypto-hashes/blob/zkvm/sha2/src/lib.rs) for hasher instantiation. So, having that, we (I believe) want to use both sha512 and blake2s sphinx precompiles integrated via "unified" dependency which is currently [zkvm](https://github.com/argumentcomputer/RustCrypto-hashes/tree/zkvm) branch, there is no definition there of `Sha512Trunc256` publicly exposed

Comment (wwared) [reply kadena/core/src/crypto/hash/sha512.rs:10]: I see, it's an upstream `sha2` type change, thanks!

File: kadena/Cargo.lock
Comment (tchataigner) [inline kadena/Cargo.lock:1]: An update of the lock file can make sense but a deletion might be a tad too violent :smile:

**Review by tchataigner — CHANGES_REQUESTED** [review]

File: kadena/Cargo.lock
Comment (storojs72) [reply kadena/Cargo.lock:1]: Sorry, that was accidentally. Returned it back

**Review by wwared — COMMENTED** [review]
I think the code in this PR is good to merge, except we need to merge https://github.com/argumentcomputer/sphinx/pull/176 first and point the `Cargo.toml` back at `dev` instead of `artem/blake2s`

File: kadena/programs/longest-chain/Cargo.toml [hunk 2]
```diff
@@ -8,8 +8,4 @@ license = "Apache-2.0"
 [dependencies]
 sphinx-zkvm = { git = "https://github.com/argumentcomputer/sphinx", branch = "dev" }
 kadena-lc-core = { path = "../../core", package = "kadena-lc-core" }
-
-[patch.crates-io]
-# Sphinx patch
-#tiny-keccak = { git = "https://github.com/sp1-patches/tiny-keccak", branch = "patch-v2.0.2" }
-sha2 = { git = "https://github.com/sp1-patches/RustCrypto-hashes", branch = "patch-v0.10.8" }
```
Comment (wwared) [inline kadena/programs/longest-chain/Cargo.toml:15]: I think we need to keep this `[patch.crates-io]` section, though we can point it at our own fork (`zkvm` branch) instead of the `sp1-patches` one

File: kadena/programs/benchmarks/block-header-hashing/Cargo.toml [hunk 3]
```diff
@@ -8,8 +8,4 @@ license = "Apache-2.0"
 [dependencies]
 sphinx-zkvm = { git = "https://github.com/argumentcomputer/sphinx", branch = "dev" }
 kadena-lc-core = { path = "../../../core", package = "kadena-lc-core" }
-
-[patch.crates-io]
-# Sphinx patch
-#tiny-keccak = { git = "https://github.com/sp1-patches/tiny-keccak", branch = "patch-v2.0.2" }
-sha2 = { git = "https://github.com/sp1-patches/RustCrypto-hashes", branch = "patch-v0.10.8" }
```
Comment (wwared) [inline kadena/programs/benchmarks/block-header-hashing/Cargo.toml:15]: Same here

**Comment by storojs72** [comment]
On hold for a while as it requires new sphinx release

**Comment by storojs72** [comment]
Closing in favour of https://github.com/argumentcomputer/zk-light-clients/pull/256
